extensions/mqtt/processors/PublishMQTT.h

```diff
@@ -79,7 +79,7 @@ class PublishMQTT : public processors::AbstractMQTTProcessor {
       if (flow_size_ < max_seg_size_)
         max_seg_size_ = flow_size_;
       std::vector<unsigned char> buffer;
-      buffer.reserve(max_seg_size_);
+      buffer.resize(max_seg_size_);
```

Comment:
Does this change belong here?
If so, moving this to the constructor is more optimal:
```c++
std::vector<unsigned char> buffer(max_seg_size_);
```

Reply:
@adamdebreceni suggested using resize instead of reserve, and I changed it in all occurrences. I'll move it to constructor.

Reply:
Updated in [48ec7a](https://github.com/apache/nifi-minifi-cpp/pull/979/commits/48ec7a2ebb6d109c147f2a3b83aa2ae8996acfa2)